### PR TITLE
add draft for test doc

### PR DESCRIPTION
**Pull request by mjudeikis**
add sub-document for how to work with tests in this repo

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# OpenShift on Azure testing
+
+OpenShift on Azure project is using few testing methods. 
+Some of the test code base is being reused from other projects.
+
+PR commands to execute tests:
+```
+/test e2e - run native e2e tests
+/test unit - run unit tests
+/test build - run container build tests
```
Comment (kargakis) [inline docs/testing.md]: this is no more

File: docs/testing.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# OpenShift on Azure testing
+
+OpenShift on Azure project is using few testing methods.
```
Comment (pweil-) [inline docs/testing.md]: "using a few"

File: docs/testing.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+/test e2e - run native e2e tests
+/test unit - run unit tests
+/test build - run container build tests
+/test upgrade - run upgrade cluster tests
+/test conformance - run origin conformance tests
+```
+
+## Unit test
+
+Packages under `pkg` directory contains their individual unit tests.
```
Comment (pweil-) [inline docs/testing.md]: s/contains/contain

File: docs/testing.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+To run all unit tests locally, execute: `make unit`.
+To run individual test:
+`go test github.com/openshift/openshift-azure/pkg/tls`
+
+## OpenShift E2E tests
+
+Project has its own End-2-End testing test suite. You can find it under:
```
Comment (pweil-) [inline docs/testing.md]: s/Project/The project

File: docs/testing.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+`test/e2e/`. It uses [ginko](https://github.com/onsi/ginkgo) and [gomega](https://github.com/onsi/gomega).
+To run E2E tests locally, execute: `make e2e`
+
+## OpenShift Origin Conformance tests
+
+For the generic OpenShift functionality we use [OpenShift Origin](https://github.com/openshift/origin)
```
Comment (pweil-) [inline docs/testing.md]: "we use the"

File: docs/testing.md (hunk 4 above)
Comment (jim-minter) [reply docs/testing.md]: end-to-end

File: docs/testing.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+/test unit - run unit tests
+/test build - run container build tests
+/test upgrade - run upgrade cluster tests
+/test conformance - run origin conformance tests
+```
+
+## Unit test
+
+Packages under `pkg` directory contains their individual unit tests.
+To run all unit tests locally, execute: `make unit`.
```
Comment (jim-minter) [inline docs/testing.md]: "or `go test ./...`" ?

File: docs/testing.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+To run individual test:
+`go test github.com/openshift/openshift-azure/pkg/tls`
```
Comment (jim-minter) [inline docs/testing.md]: To run a single package's tests:
`go test ./pkg/tls`

To run a subset of tests in a package:
`go test -run TestFoo ./pkg/tls`

File: docs/testing.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+git clone https://github.com/<gh_username>/origin $GOPATH/src/github.com/openshift/origin
+cd $GOPATH/src/github.com/openshift/origin
+# build extended test
+make build-extended-test
+# export kubeconfig
+export KUBECONFIG=$(pwd)/_data/_out/admin.kubeconfig
+# run test via wrapper
+TEST_ONLY=1 test/extended/core.sh -test.timeout=2h -ginkgo.focus="${TEST_FOCUS}" \
+-report-dir /tmp/_data/junit
+```
```
Comment (jim-minter) [inline docs/testing.md]: Please could you add another section on running e2e tests? TL;DR:
* deploy a cluster
* `go test -tags e2e ./test/e2e` or `make e2e`

Comment (mjudeikis) [reply docs/testing.md]: done

**Comment by mjudeikis** [comment]
ping @jim-minter @pweil-

**Pull request merged**